protocols/kad/src/handler.rs

```diff
@@ -774,6 +779,15 @@ where
             return Poll::Ready(event);
         }
 
+        let num_in_progress_outbound_substreams =
+            self.outbound_substreams.len() + self.num_requested_outbound_streams;
+        if MAX_NUM_SUBSTREAMS.saturating_sub(num_in_progress_outbound_substreams) > 0 {
```

Comment:
```suggestion
        if num_in_progress_outbound_substreams < MAX_NUM_SUBSTREAMS {
```

Why not just compare the two numbers directly?